tools/fast/series/mining_once_from_ctx.go

```diff
@@ -0,0 +1,19 @@
+package series
+
+import (
+	"context"
+)
+
+// CKMiningOnce is the key used to store the MiningOnceFunc in a context
+var CKMiningOnce = struct{}{}
```

Comment:
Name suggestions? Probably can be private and export a function to set the value on an accepted context.